diff --git a/lib/main.dart b/lib/main.dart
index 6ad9f5e..d6ed5b3 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -15,13 +15,14 @@ import 'package:flutter/material.dart';
 import 'system.dart';
 import 'dart:io';
 
-void main() {
+var defaultTheme;
+
+void main() async {
   loadConfig();
+  defaultTheme = await getSystemTheme();
   runApp(TextEditorApp());
 }
 
-var defaultTheme = getSystemTheme();
-
 class TextEditorApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
diff --git a/lib/system.dart b/lib/system.dart
index a2976c6..a6843de 100644
--- a/lib/system.dart
+++ b/lib/system.dart
@@ -38,7 +38,7 @@ Future<String> _loadConfigAsset() async {
       .readAsStringSync();
 }
 
-Future loadConfig() async {
+Future<String> loadConfig() async {
   String jsonString = await _loadConfigAsset();
   final jsonResponse = json.decode(jsonString);
   Configuration config = new Configuration.fromJson(jsonResponse);
@@ -47,10 +47,10 @@ Future loadConfig() async {
   return value;
 }
 
-ThemeMode getSystemTheme() {
+Future<ThemeMode> getSystemTheme() async {
   final theme = await loadConfig();
   print(loadConfig());
-  if ("dark" == "dark") {
+  if (theme == "dark") {
     print("DARK");
     return ThemeMode.dark;
   } else {
